go/systests/team_invite_test.go

```diff
@@ -199,3 +202,135 @@ func TestTeamInviteResetNoKeys(t *testing.T) {
+	defer tt.cleanup()
+
+	alice := tt.addUser("alice")
+	bob := tt.addUser("bob")
+
+	displayNameRooter := strings.Join([]string{alice.username, bob.username + "@rooter"}, ",")
+
+	team, err := alice.lookupImplicitTeam(true /*create*/, displayNameRooter, false /*isPublic*/)
+	require.NoError(t, err)
+
+	t.Logf("Created implicit team %q -> %s\n", displayNameRooter, team)
+
+	// Bob has not proven rooter yet, so this will create a new, separate team.
+	displayNameKeybase := strings.Join([]string{alice.username, bob.username}, ",")
+	team2, err := alice.lookupImplicitTeam(true /*create*/, displayNameKeybase, false /*isPublic*/)
+	require.NoError(t, err)
+	require.NotEqual(t, team, team2)
+
+	t.Logf("Created implicit team %q -> %s\n", displayNameKeybase, team2)
+
+	bob.proveRooter()
+
+	alice.kickTeamRekeyd()
+
+	alice.waitForTeamIDChangedGregor(team, keybase1.Seqno(2))
+
+	// Display name with rooter name is no longer usable.
+	_, err = alice.lookupImplicitTeam(false /*create*/, displayNameRooter, false /*isPublic*/)
+	require.Error(t, err)
+
+	// "LookupOrCreate" rooter name should fail as well.
+	_, err = alice.lookupImplicitTeam(true /*create*/, displayNameRooter, false /*isPublic*/)
+	require.Error(t, err)
+
+	// Clients should refer to this team using comma-separated usernames.
+	_, err = alice.lookupImplicitTeam(false /*create*/, displayNameKeybase, false /*isPublic*/)
+	require.NoError(t, err)
+}
+
+func TestImpTeamWithMultipleRooters(t *testing.T) {
+	tt := newTeamTester(t)
+	defer tt.cleanup()
+
+	alice := tt.addUser("ali")
+	bob := tt.addUser("bob")
+	charlie := tt.addUser("cha")
+
+	// Both teams include social assertions, so there is no definitive conflict winner.
+	displayNameRooter1 := strings.Join([]string{alice.username, bob.username, charlie.username + "@rooter"}, ",")
+	displayNameRooter2 := strings.Join([]string{alice.username, bob.username + "@rooter", charlie.username}, ",")
+
+	team1, err := alice.lookupImplicitTeam(true /*create*/, displayNameRooter1, false /*isPublic*/)
+	require.NoError(t, err)
+
+	team2, err := alice.lookupImplicitTeam(true /*create*/, displayNameRooter2, false /*isPublic*/)
+	require.NoError(t, err)
+
+	require.NotEqual(t, team1, team2)
+
+	// proveRooter has a problem where some code path relies on global
+	// libkb.G context, so we need to change it before for each user.
+	libkb.G = bob.tc.G
+	bob.proveRooter()
+	libkb.G = charlie.tc.G
+	charlie.proveRooter()
+
+	alice.kickTeamRekeyd()
+
+	toSeqno := keybase1.Seqno(2)
+	var winningTeam keybase1.TeamID
+	for i := 0; i < 10; i++ {
+		select {
+		case arg := <-alice.notifications.rotateCh:
+			t.Logf("membership change received: %+v", arg)
+			if (arg.TeamID.Eq(team1) || arg.TeamID.Eq(team2)) && arg.Changes.MembershipChanged && !arg.Changes.KeyRotated && !arg.Changes.Renamed && arg.LatestSeqno == toSeqno {
+				t.Logf("change matched with %q", arg.TeamID)
+				winningTeam = arg.TeamID
+				return
+			}
+		case <-time.After(1 * time.Second):
```

Comment:
Worried about this on CI, but if it's currently passing I guess it is ok.

Reply:
this is copied from `waitForTeamIDChangedGregor` in `teams_test.go` but with a change that it waits for a notification from either of two given teamid instead of just one.

Reply:
ok!